src/hipscat/pixel_math/README.md

```diff
@@ -153,3 +153,44 @@ faces).
 
 #### Implementation
 Pretty straightforward implementation of the algorithm above.
+
+## Margin Bounding
+After constraining our data points using the `get_margin` code in `pixel_margins`, we then move on to our more accurate bound checking by building bounding boxes that include a region approximately one `margin_threshold` wide around the original healpixel.
+
+To get this bounding box, we
+- find a `scale` factor to apply to the original healpixel boundaries that increases the resolution by one `margin_threshold`
+- sample a set of points along the boundary of a healpixel
+- apply an affine transformation to these points
+
+resulting in a box that covers a border region of one `margin_threshold` around the original healpixel.
+
+We then can input these points into an astropy `regions.PolygonPixelRegion` object that we can the use to quickly check different datapoints against, resulting in a set of data for the final `neighbors.parquet` file.
+
+### get_margin_scale
+Finds the scale factor that we want to use to scale up the healpixel bounds by to include the neighbor margin.
+
+#### Algorithm
+- get the resolution of our `pixel_order` (sqrt of the pixel area)
+- add the `margin_threshold` to the resolution and square it.
+- divide this new area against the original pixel area to find the scale factor.
+
+### get_margin_bounds_and_wcs
+Given a healpixel and a scale factor, generate a `regions.PolygonPixelRegion` polygon and an `astropy.wcs.WCS` object containing the points of the healpixel scaled around the centroid by a factor of `scale`. Used in conjunction with `get_margin_scale` to perform an affine transform on a set of coordinates sampled from the boundaries of a healpixel.
+
+By returning it as a pixel region along with a wcs object, we can quickly check data points against our polygon.
+
+In the case where `pixel_order` is less than 2, we divide the polygon into 4 or 16 different polygon regions (orders **0** and **1** respectively), each with their own `WCS` object. We do this because `PolygonPixelRegions` start to break down with large bounding boxes at the granular coordinate spaces that we're using.
```

Comment:
Won't this then give you the boundary points that are inside the original (larger) pixel?

Reply:
yes, if you check a point that lies within the original pixel against the bounding box it will return true, however from the constraining step (where we collect data from neighboring smaller order pixels) we only run `check_margin_bounds` against data that is by definition not in the original pixel.

I played around with trying to create bounding boxes that just cover the margin region itself, but abandoned it for 3 reasons:

- it adds a lot of undue complexity to the code when we don't really need to as we'll never run the margin bound checks against the original pixel data
- by trying to build another boundary along the inner margin region, we introduce a larger error when transitioning from spherical coordinates to cartesian ones 
- i generally wanted to err on the side of including more data than excluding, as the purpose of margin caching is to prevent data loss (especially with data points super close to the boundaries)